tsnet/tsnet.go

```diff
@@ -829,20 +831,22 @@ func (s *Server) ListenTLS(network, addr string) (net.Listener, error) {
 		return nil, errors.New("tsnet: you must enable HTTPS in the admin panel to proceed. See https://tailscale.com/s/https")
 	}
 
-	lc, err := s.LocalClient() // do local client first before listening.
-	if err != nil {
-		return nil, err
-	}
-
 	ln, err := s.listen(network, addr, listenOnTailnet)
 	if err != nil {
 		return nil, err
 	}
 	return tls.NewListener(ln, &tls.Config{
-		GetCertificate: lc.GetCertificate,
+		GetCertificate: s.getCert,
 	}), nil
 }
 
+func (s *Server) getCert(hi *tls.ClientHelloInfo) (*tls.Certificate, error) {
+	if s.getCertForTesting != nil {
+		return s.getCertForTesting(hi)
+	}
+	return s.localClient.GetCertificate(hi)
```

Comment:
`lc` can be nil if not server not started

```suggestion
    lc, err := s.LocalClient()
    if err != nil {
        return nil, err
    }
	return lc.GetCertificate(hi)
```

Reply:
all callers were already starting the server, but yeah its fragile. Went with your suggestion.